### PR TITLE
Simplify `var/` folder and similar

### PR DESCRIPTION
**Pull request by giorgiosironi**
- Remove `setgid` bit complexity
- Remove obsolete service stopping state
- Remove git repository absence state
- Avoid removing file that does not exists anymore

### CODE REVIEW

File: salt/journal/init.sls [hunk 1]
```diff
@@ -74,16 +68,15 @@ assets-nginx-configuration:
...
             - journal-folder
 
-# files and directories must be readable and writable by both elife and www-data
-# they are both in the www-data group, but the g+s flag makes sure that
-# new files and directories created inside have the www-data group
+# files and directories must be writable by www-data,
+# the user the container is running as
 logs-directory:
     file.directory:
         - name: /srv/journal/var/logs
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: Probably don't actually need `var` anymore?

Comment (giorgiosironi) [reply salt/journal/init.sls]: Just keeping it consistent with the filesystem inside the container to prevent confusion

Comment (giorgiosironi) [reply salt/journal/init.sls]: Also migrating the path of a volume on the host is a pain

Comment (thewilkybarkid) [reply salt/journal/init.sls]: Guess that explains https://github.com/elifesciences/journal-formula/blob/d06cdff4c0919bb302489489ed4fc7ed515d41a9/salt/journal/init.sls#L58-L59

Comment (giorgiosironi) [reply salt/journal/init.sls]: At least that's readonly so cleanup is just removing the old file with another state; but `logs/` is writable so can't move it without stopping the containers preemptively, move files, restart.

**Pull request merged**